builtin/providers/aws/resource_aws_rds_cluster_parameter_group_test.go

```diff
@@ -0,0 +1,301 @@
+			t.Fatalf("Expected the DB Cluster Parameter Group Name to trigger a validation error")
+		}
+	}
+}
+
+func testAccCheckAWSDBClusterParameterGroupDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_rds_cluster_parameter_group" {
+			continue
+		}
+
+		// Try to find the Group
+		resp, err := conn.DescribeDBClusterParameterGroups(
+			&rds.DescribeDBClusterParameterGroupsInput{
+				DBClusterParameterGroupName: aws.String(rs.Primary.ID),
+			})
+
+		if err == nil {
+			if len(resp.DBClusterParameterGroups) != 0 &&
+				*resp.DBClusterParameterGroups[0].DBClusterParameterGroupName == rs.Primary.ID {
+				return fmt.Errorf("DB Cluster Parameter Group still exists")
+			}
+		}
+
+		// Verify the error
+		newerr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if newerr.Code() != "DBParameterGroupNotFound" {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSDBClusterParameterGroupAttributes(v *rds.DBClusterParameterGroup) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if *v.DBClusterParameterGroupName != "cluster-parameter-group-test-terraform" {
+			return fmt.Errorf("bad name: %#v", v.DBClusterParameterGroupName)
+		}
+
+		if *v.DBParameterGroupFamily != "aurora5.6" {
+			return fmt.Errorf("bad family: %#v", v.DBParameterGroupFamily)
+		}
+
+		if *v.Description != "Test cluster parameter group for terraform" {
+			return fmt.Errorf("bad description: %#v", v.Description)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSDBClusterParameterGroupExists(n string, v *rds.DBClusterParameterGroup) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No DB Cluster Parameter Group ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).rdsconn
+
+		opts := rds.DescribeDBClusterParameterGroupsInput{
+			DBClusterParameterGroupName: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeDBClusterParameterGroups(&opts)
+
+		if err != nil {
+			return err
+		}
+
+		if len(resp.DBClusterParameterGroups) != 1 ||
+			*resp.DBClusterParameterGroups[0].DBClusterParameterGroupName != rs.Primary.ID {
+			return fmt.Errorf("DB Cluster Parameter Group not found")
+		}
+
+		*v = *resp.DBClusterParameterGroups[0]
+
+		return nil
+	}
+}
+
+const testAccAWSDBClusterParameterGroupConfig = `
+resource "aws_rds_cluster_parameter_group" "bar" {
+	name = "cluster-parameter-group-test-terraform"
+	family = "aurora5.6"
+	description = "Test cluster parameter group for terraform"
+	parameter {
+	  name = "character_set_server"
+	  value = "utf8"
+	}
+	parameter {
+	  name = "character_set_client"
+	  value = "utf8"
+	}
+	parameter{
+	  name = "character_set_results"
+	  value = "utf8"
+	}
+	tags {
+		foo = "bar"
+	}
+}
```

Comment:
Small style nit - can you format these how `hclfmt` would? (two space indents, aligned equals)